### DISCUSSION TITLE
Does jaeger remote sampler need to look at Links?

### DISCUSSION THREAD

**Discussion by anuraaga** [open-telemetry/opentelemetry-java#2874] [Q&A]
Currently, our jaeger remote sampler looks at links in the rate limiting sampler

https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/RateLimitingSampler.java#L65

It doesn't for the probabilistic sampler. I don't think it does upstream either

https://github.com/jaegertracing/jaeger-client-java/blob/master/jaeger-core/src/main/java/io/jaegertracing/internal/samplers/RateLimitingSampler.java#L49

So I suspect it's ok to remove this `Link`-based sampling to clean things up a bit - @yurishkuro can you confirm?

**Comment by yurishkuro** [comment]
Historically, Jaeger samplers only had access to trace id and operation name. In Go and Node.js they have access to the full span and start/finish/setTag life cycle events. They theoretically do have access to Span References, but we never had a use case for that.

**Reply by yurishkuro** [reply]
Also, rate limiting sampler is in the category of "primitive" samplers, they don't look at any span details at all.
